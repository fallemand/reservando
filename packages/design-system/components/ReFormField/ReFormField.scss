@import "../../styles/variables";

$padding-y: 0.5em;
$padding-x: 0.75em;

.re-form-field {
  display: flex;
  flex-direction: column;

  &__error {
    color: $red-feedback;
    font-size: 0.875em;
  }
}
