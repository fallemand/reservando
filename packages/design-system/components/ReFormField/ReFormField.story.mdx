import { Meta, Story, Canvas, Props, Description, ArgsTable } from "@storybook/addon-docs/blocks";
import { ReFormField, ReInput } from "@reservando/design-system";

<Meta title="Components/ReFormField" component={ReFormField} />

# ReFormField

<Description of={ReFormField} />

<Canvas>
  <Story name="basic">
    {{
      components: { ReFormField, ReInput },
      data() {
        return {
          text: "",
        };
      },
      template: `
      <div style="max-width: 400px;">
        <ReFormField error="Please complete the value">
          <ReInput ref="input" v-model="text" required />
        </ReFormField>
      </div>
    `,
    }}
  </Story>
</Canvas>

## How to use

```html
<script lang="ts">
  import { ReFormField } from "@reservando/design-system";
</script>
```

<ArgsTable of={ReFormField} />
