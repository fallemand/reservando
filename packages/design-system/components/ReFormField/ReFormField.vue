<template>
  <div class="re-form-field">
    <slot />
    <ReTransition name="height">
      <p v-show="error" class="re-form-field__error">
        {{ error }}
      </p>
    </ReTransition>
  </div>
</template>

<script lang="ts">
import { defineComponent } from "vue";
import ReTransition from "../ReTransition/ReTransitionGroup.vue";

/**
 * This component allows you to add `label` `hint` & `error` to any form component. <br />
 * Use it to wrap your ReInput, ReSelect, ReCheckbox, etc
 */
export default defineComponent({
  name: "ReFormField",
  components: {
    ReTransition,
  },
  props: {
    error: {
      default: null,
      type: String,
    },
  },
});
</script>

<style lang="scss" src="./ReFormField.scss" />
