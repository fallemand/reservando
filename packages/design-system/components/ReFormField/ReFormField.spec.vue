<script lang="ts">
import { shallowMount } from "@vue/test-utils";
import ReFormField from "./ReFormField.vue";

describe("ReFormField component", () => {
  const props = {
    error: "__ERROR__",
  };
  it("should render with default props", () => {
    const wrapper = shallowMount(ReFormField, {
      props,
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
</script>
