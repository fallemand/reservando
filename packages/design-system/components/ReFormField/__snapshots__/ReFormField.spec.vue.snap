// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReFormField component should render with default props 1`] = `
<div
  class="re-form-field"
>


  <re-transition-stub
    name="height"
  />
</div>
`;
